controllers/metal3.io/host_state_machine.go

```diff
@@ -585,6 +585,13 @@ func (hsm *hostStateMachine) handlePoweringOffBeforeDelete(info *reconcileInfo)
 			return skipToDelete()
 		}
 	case actionError:
+		if r.NeedsRegistration() && hsm.haveCreds {
+			if hsm.Host.Status.ErrorCount > 3 {
+				info.log.Info("Giving up on host power off after 3 attempts.")
+				return skipToDelete()
+			}
```

Comment:
Maybe if we did something like

```
return recordActionFailure(info, metal3api.RegistrationError, r.err.Error())
```

here (after the if block) then we wouldn't need to change the behaviour of every error return point (most of which are likely transient errors).

Reply:
(but see main comment - on reflection I don't think we need to count at this point at all)